<a href="https://colab.research.google.com/github/ashifka/ashif/blob/main/ASHIF_KA(Case_Study_on_Regression).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## **LINEAR REGRESSION**

In [12]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_absolute_error
data = pd.read_csv('/content/sample_data/car_age_price.csv')
data.head(10)

Year   Price
0  2018  465000
1  2019  755000
2  2019  700000
3  2018  465000
4  2018  465000
5  2015  350000
6  2016  425000
7  2017  575000
8  2017  509999
9  2017  575000

In [13]:
data.isnull().sum()

Year     0
Price    0
dtype: int64

In [2]:
# Splitting the data into training and testing sets
X = data[['Year']]
y = data['Price']
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

In [3]:
# Creating and fitting the linear regression model
linear_reg = LinearRegression()
linear_reg.fit(X_train, y_train)

LinearRegression()

In [4]:
# Predicting on the test set
y_pred_linear = linear_reg.predict(X_test)
y_pred_linear


array([600775.91252081, 505558.77690466, 553167.34471273, 553167.34471273,
       553167.34471273, 410341.6412885 , 505558.77690466, 553167.34471273,
       600775.91252081, 600775.91252081, 315124.50567235, 505558.77690466,
       410341.6412885 , 648384.48032889, 553167.34471273, 600775.91252081,
       315124.50567235, 410341.6412885 , 505558.77690466, 505558.77690466,
       505558.77690466, 505558.77690466, 505558.77690466])

In [5]:
y_test

40     525000
65     495000
4      465000
47     450000
42     550000
69     300000
26     550000
11     500000
10     500000
44     550000
73     320000
55     550000
68     465000
97     640000
0      465000
18     500000
53     345000
80     390000
103    375000
91     520000
36     550000
12     500000
31     550000
Name: Price, dtype: int64

In [6]:
# Calculating MAE for linear regression
mae_linear = mean_absolute_error(y_test, y_pred_linear)
print("Linear Regression MAE:", mae_linear)

Linear Regression MAE: 53534.77957001264


In [7]:
# Predicting the price of a 2022 model
predicted_2022_price_linear = linear_reg.predict([[2022]])
print("Predicted price for 2022 (Linear Regression):", predicted_2022_price_linear[0])

Predicted price for 2022 (Linear Regression): 743601.6159450412


/usr/local/lib/python3.10/dist-packages/sklearn/base.py:439: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


## **LASSO REGRESSION**

In [8]:
from sklearn.linear_model import Lasso

# Creating and fitting the Lasso regression model
lasso_reg = Lasso(alpha=0.1)  # You can adjust alpha for regularization strength
lasso_reg.fit(X_train, y_train)

# Predicting on the test set
y_pred_lasso = lasso_reg.predict(X_test)
y_pred_lasso


array([600775.81201603, 505558.75884159, 553167.28542881, 553167.28542881,
       553167.28542881, 410341.70566715, 505558.75884159, 553167.28542881,
       600775.81201603, 600775.81201603, 315124.65249272, 505558.75884159,
       410341.70566715, 648384.33860324, 553167.28542881, 600775.81201603,
       315124.65249272, 410341.70566715, 505558.75884159, 505558.75884159,
       505558.75884159, 505558.75884159, 505558.75884159])

In [9]:
y_test

40     525000
65     495000
4      465000
47     450000
42     550000
69     300000
26     550000
11     500000
10     500000
44     550000
73     320000
55     550000
68     465000
97     640000
0      465000
18     500000
53     345000
80     390000
103    375000
91     520000
36     550000
12     500000
31     550000
Name: Price, dtype: int64

In [10]:
# Calculating MAE for Lasso regression
mae_lasso = mean_absolute_error(y_test, y_pred_lasso)
print("Lasso Regression MAE:", mae_lasso)

Lasso Regression MAE: 53534.73464390959


In [11]:
# Predicting the price of a 2022 model
predicted_2022_price_lasso = lasso_reg.predict([[2022]])
print("Predicted price for 2022 (Lasso Regression):", predicted_2022_price_lasso[0])

Predicted price for 2022 (Lasso Regression): 743601.3917776793


/usr/local/lib/python3.10/dist-packages/sklearn/base.py:439: UserWarning: X does not have valid feature names, but Lasso was fitted with feature names
  warnings.warn(


FROM THE MAE(MEAN ABSOLUTE ERROR) OF LINEAR AND LASSO REGRESSION
*   Lasso Regression MAE: 53534.73464390959
*   Linear Regression MAE: 53534.77957001264

---


LESSER THE MAE IS BETTER PERFORMANCE.IN HERE BOTH ARE ALMOST SAME,BUT LASSO REGRESSION IS SLIGHTLY LESSER THAN LINEAR REGRESSION......
HENCE LETS CONCLUDE WITH THAT LASSO REGRESSION IS BETTER PERFORMANCE


